mgmtsystem/ec2.py

```diff
@@ -450,14 +454,14 @@ def remove_host_from_cluster(self, hostname):
 
     def create_s3_bucket(self, bucket_name):
         self.logger.info("Creating bucket: {}".format(bucket_name))
-        self.s3_connection.create_bucket(bucket_name)
+        return self.s3_connection.create_bucket(Bucket=bucket_name, CreateBucketConfiguration={
+            'LocationConstraint': self.kwargs.get('region')})
 
-    def upload_file_to_s3_bucket(self, bucket_name, file_path, key_name):
-        bucket = self.s3_connection.get_bucket(bucket_name)
+    def upload_file_to_s3_bucket(self, bucket_name, file_path, file_name):
+        bucket = self.s3_connection.Bucket(bucket_name)
         self.logger.info("uploading file {} to bucket: {}".format(file_path, bucket_name))
-        key = boto.s3.key.Key(bucket, key_name)
-        with open(file_path) as f:
-            key.set_contents_from_file(f)
+        with open(file_path):
+            bucket.upload_file(file_path, file_name)
         self.logger.info("Success: uploading file completed")
```

Comment:
The success here - how is it checked? What does the upload_file return? Does it throw an exception?